### PR TITLE
Model refactor

### PR DESCRIPTION
**Pull request by thepwagner**
The end-game is letting a SpaceApp change between regions; the previos
BaseModelObject commit started down that direction, but I didn't like
how `spacel.model.*` started to look/feel.

Opting to introduce new classes for building `Orbit`/`SpaceApp` from a
JSON struct, looking towards #40 and #41 where another layer may mixing
files/directory contents.

Also adding in a `.valid` property to both, for local validation of
parameters/state (WEBOPS-403?)

Refactoring those model objects breaks _everything_; but this is
probably the last missing "some day I'll add that" feature before 1.0.0.

### CODE REVIEW

**Comment by thepwagner** [comment]
API impacts:
https://github.com/pebble/spacel-provision/blob/master/sample/orbit/develop.json
`.defaults` is renamed to `.all` (potato/potatoe difference).

https://github.com/pebble/spacel-provision/blob/master/sample/app/http-env-echo.json every key excepting `.name` is nested under a `.all`, and can be overwritten by `.${region}` (i.e. `.us-west-2.services`)

Is the `.all`/`.default` layer sane? Or should "default if not over-written" values just live in the root? (for both?)

**Comment by DeviaVir** [comment]
@thepwagner I think the `.all/.default` pattern makes sense for both here, otherwise we'll have `region` logic at the same height as `app` logic, this prevents that and gives a nice clear distinction.

Curious how this could work out with R53, could turn out real nice.

**Comment by thepwagner** [comment]
@DeviaVir clear for review now; includes #59 under the assumption that that will land first.

File: src/spacel/model/json/base.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    'services'
+}
+
+TRUTH_VALUES = {
+    True,
+    '1',
+    'yes',
+    'enabled',
+    'true',
+    'please',  # en_CA support
```
Comment (DeviaVir) [inline src/spacel/model/json/base.py]: 🇨🇦

File: src/spacel/provision/app/db/rds.py [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
 from spacel.provision import clean_name, bool_param
 from spacel.provision.app.db.base import BaseTemplateDecorator
 
 logger = logging.getLogger('spacel.provision.rds.factory')
 
+POSTGRES_PORT = 5432
+POSTGRES_VERSION = '9.5.2'
+
 DEFAULT_VERSIONS = {
     'mysql': '5.7.10',
```
Comment (DeviaVir) [inline src/spacel/provision/app/db/rds.py]: Why not do the same for `mysql` if you're already touching this?

**Comment by thepwagner** [comment]
I've run the suite 1-by-1 while updating them, which feels "good enough" for merge.
I don't want to tie up `sl-test@us-east-1` and/or my machine for 2h today to run them; I'll make a pass this weekend when I can set-it-and-forget-it and not eat up potentially productive time.

**Comment by thepwagner** [comment]
_Did_ break 2 integration tests; fixed in https://github.com/pebble/spacel-provision/pull/62

**Pull request merged**